axum/src/extract/path/mod.rs

```diff
@@ -380,7 +380,14 @@ impl fmt::Display for ErrorKind {
 pub struct FailedToDeserializePathParams(PathDeserializationError);
 
 impl FailedToDeserializePathParams {
+    /// Get the underlying error kind.
+    pub fn kind(&self) -> ErrorKind {
+        self.0.kind.clone()
+    }
```

Comment:
Any reason to not return a reference here? I guess it means we change the internals of `FailedToDeserializePathParams` but maybe that's okay

```suggestion
    pub fn kind(&self) -> &ErrorKind {
        &self.0.kind
    }
```

Reply:
If we commit to `FailedToDeserializePathParams` containing `ErrorKind`, do we keep `into_kind` or deprecate / later remove it? If we keep the `Clone` impl I added for making the current implementation of this method possible, `.kind().clone()` would be semantically equivalent to the current `.into_kind()` (though theoretically less efficient).

Reply:
Ping @davidpdrsn

Reply:
I'm fine with committing to `ErrorKind` for now.

Reply:
Err, I don't see how that answers my question.. I guess I will just do the easiest thing, keep `into_kind`, remove the `Clone` impl and not deprecate anything.

Reply:
Ah sorry. I thought you were asking about committing to having an `ErrorKind` inside `FailedToDeserializePathParams`, not whether to keep `into_kind` and `kind`. I'm fine with keeping both methods, like you've implemented it here.